documentation/source/user_section/tutorials.rst

```diff
@@ -1,9 +1,33 @@
-.. _sct_courses:
+.. _tutorials:
 
-Courses
-#######
+Tutorials and Courses
+#####################
 
-We organize **free** SCT courses, each year after the ISMRM conference. If you’d like to be added to the mailing list, please visit `this link <https://docs.google.com/forms/d/e/1FAIpQLSdvsXwLRj07bx85O2wPM9ut1sC-20dFKdTVe-OckyvkfncV3Q/viewform>`_.
+Tutorial pages
+**************
+
+We provide the following hands-on tutorials for SCT's command-line tools. We recommend that you complete each tutorial in sequence using the "Next" and "Previous" buttons on each page.
+
+.. toctree::
+   :maxdepth: 2
+
+   tutorials/1
+   tutorials/2
+   tutorials/3
+   tutorials/4
+   tutorials/5
```

Comment:
### Splitting RTD into multiple pages

Based on a suggestion from the previous SCT meeting, in f3ae97d I've split up each tutorial page into smaller numbered subpages, with the goal to make it more manageable for presentations. The idea here is that we would navigate each page with the "Next" and "Previous" sections.

I've marked the commit as "`(WIP)`" because this is a rough sketch of the layout; there are still some wording/formatting tweaks needed to make sure everything flows. But before I commit more time to cleaning things up, I wanted to ask for feedback on the layout to make sure this is even something we want.

Reply:
Some feedback/questions:
- I _think_ I do prefer when tutorials are split up into smaller sections, but I also see how it could be overwhelming to have so many headings on the TOC:
  
  ![image](https://user-images.githubusercontent.com/2482071/120088841-a63aec80-c0c2-11eb-8ce4-a85ba39b389e.png)

- Related to my previous point, with the current RTD template it is a bit difficult to navigate the TOC, but once https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3355 is in, it might become easier (thereby lowering the weight on the previous point)
- On our end, I am wondering how much more work it is to split tutorials into smaller chunks? Also, anticipating maintenance for functions that will eventually be deprecated (eg: deepseg_sc vs. deepseg).
- Unrelated to option 🎉  or 🚀: I am wondering if we need to keep the numbering of tutorials, ie: remove "tutorial 1:", "tutorial 2:", etc.
- Also unrelated to the two options: we might want to make vertebral labeling its separate tutorial, because it is not uncommon that ppl want to label vertebrae without registering to the PAM50.

Reply:
>  I also see how it could be overwhelming to have so many headings on the TOC:

Registration was the biggest section in the course, so Tutorial 2 would certainly be the longest. Though, if we made vertebral labeling its own section (as you suggest), then this might be less of a problem?

> On our end, I am wondering how much more work it is to split tutorials into smaller chunks? 

For me personally, splitting it up makes writing a little easier. The structure requires me to make sure each section is focused and self-contained, which I found beneficial for organizing the material.

> Also, anticipating maintenance for functions that will eventually be deprecated (eg: deepseg_sc vs. deepseg).

True. Though, this seems necessary for any tutorial format we choose. (e.g. the slide decks would have needed to be updated as well)

> Related to my previous point, with the current RTD template it is a bit difficult to navigate the TOC, but once #3355 is in, it might become easier (thereby lowering the weight on the previous point)

Thank you for pointing this out. Tomorrow, I can briefly copy the theme changes from that PR so we can double-check how this will look with the new theme. I'm especially curious about the "mini TOC" in the top-right of that theme.

> I am wondering if we need to keep the numbering of tutorials, ie: remove "tutorial 1:", "tutorial 2:", etc.

Do you mean all numbering (1.1, 2.3, etc.) or just the tutorial title numbering ("tutorial 1:", "tutorial 2")?

Reply:
> Do you mean removing all numbering (1.1, 1.2, 2.1, etc.) or just the tutorial title numbering ("tutorial 1:", "tutorial 2")?

i meant the title numbering, but i guess if we remove that we should also remove the sub-numbering 2.1 etc.
maybe that’s a bad idea

Reply:
> if we insert tutorials in between it will force the iteration of all subsequent tutorials

Inserting tutorials in between is tricky, because the tutorials are currently expected to be run in sequence. Every step in the SCT course re-used files from the previous step:

 4 - `sct_register_multimodal` (MT) depends on output from `sct_register_to_template` (T2)
 3 - `sct_process_segmentation` (T2) also depends on output from `sct_register_to_template` (T2)
 2 - `sct_register to template` depends on output from `sct_label_vertebrae` and `sct_deepseg_sc`
 1 - `sct_label_vertebrae` depends on output from `sct_deepseg_sc`

So, the tricky part is: How do we make the tutorials stand alone, AND also let them be run in sequence.

Reply:
> Hmm... Inserting tutorials in between is tricky, because the tutorials are currently expected to be run in sequence.

What I had in mind was possibly breaking down some of them, while keeping the same sequence ordering, e.g extracting the vertebral labeling from the "register to template" tutorial. But in general yes, i agree with you.

> Right now, each tutorial is not written in a way where it can stand alone, because we would have to duplicate the earlier steps into the later tutorials... Unless we also included the intermediate output files in the example dataset, so people could "jump right in" to a later tutorial. But, then the output of earlier tutorials would already be there... quite the conundrum. 

Yes, I did think about adding the outputs of previous steps, but cons are:
- it would bring confusion (people would see the message: "file already exists, overwriting it, etc.")
- example dataset will be heavier (esp. those warp_ files are large)

I was hoping we could have standalone tutorials like in [scikit-learn](https://scikit-learn.org/stable/auto_examples/index.html), but maybe that's not such a good idea in the context of SCT, which is heavily pipeline-oriented. Maybe this is something we can bring at the next SCT meeting.

Reply:
> it would bring confusion (people would see the message: "file already exists, overwriting it, etc.")

One way to get around this could be to make the website download links for each tutorial contain the necessary intermediate files, so each tutorial can be run standalone.

Then, in the SCT course, we could switch to a course-specific dataset which is used for running through all of the tutorials in full. 

That way we get the best of both worlds... on the website, they will be standalone, but when taught as a course, they all naturally chain together.

Reply:
I like this idea. It will require a bit more maintenance on our end (ie: generate multiple downloadable packages for each tutorial), but it's not the end of the world.

Reply:
>  It will require a bit more maintenance on our end (ie: generate multiple downloadable packages for each tutorial), but it's not the end of the world.

Hmm... we are storing all of our files on GitHub currently (sct_testing_data, sct_example_data). Is it possible we could link directly to the files inside our existing test/example datasets? For example:

> This tutorial requires that you have the following files in your working directory:
>
>   * [t2_seg.nii.gz](https://github.com/spinalcordtoolbox/sct_testing_data/raw/master/t2/t2_seg-manual.nii.gz)
>   * etc.
>
> To get these files, you can either download them from the list, or you can follow the previous tutorial [link] to generate these files yourself.

Reply:
Maybe not... Realizing now that browser download managers try to unzip `.gz` archives by default. :sweat_smile: 

![Screenshot from 2021-05-31 13-59-12](https://user-images.githubusercontent.com/16181459/120228137-874e6e80-c218-11eb-933a-dd9f2b96c212.png)

Reply:
> Maybe not... Realizing now that browser download managers try to unzip .gz archives by default. 😅

ahah yeah... also we would want to keep the same consistent folder structure (data_course/t2/anat/blabla/) and if we distribute individual files i foresee issues with users putting the files in wrong directories, etc.

Reply:
> It will require a bit more maintenance on our end (ie: generate multiple downloadable packages for each tutorial), but it's not the end of the world.

I have a proposal for managing this. 

We would start by only storing the basic files necessary for the SCT course in a repo (e.g. `sct_course_data`), and create releases exactly like we've been doing in the past (e.g. `sct_course_london20.zip`).

Importantly, though, we also currently include a `batch_single_subject.sh` script that mirrors the course commands. So, we would then create a GH Actions workflow that will run the batch script, generate the intermediate files, package them up into tutorial-specific subsets, and attach the `.zip`s to each release. That way, the release page will automatically have two separate types of downloads: the main course dataset, plus the tutorial-specific datasets (which we can link to in the docs).

I'm currently experimenting with this idea [here](https://github.com/spinalcordtoolbox/sct_tutorial_data/blob/master/.github/workflows/create_release.yml).

Reply:
Update: I've finished the workflow and tested it out. 

* Sample workflow run: https://github.com/spinalcordtoolbox/sct_tutorial_data/runs/2732690858?check_suite_focus=true
* The resulting release with sub-datasets: https://github.com/spinalcordtoolbox/sct_tutorial_data/releases/tag/r20200603
* The file defines sub-datasets: https://github.com/spinalcordtoolbox/sct_tutorial_data/blob/master/tutorial-datasets.csv

Reply:
I think I have responded to all of the feedback in this thread. I plan on proofreading + responding to @ahill187's feedback about the content itself (typos/wording/layout, etc.) next.

> Also unrelated to the two options: we might want to make vertebral labeling its separate tutorial, because it is not uncommon that ppl want to label vertebrae without registering to the PAM50.

Done in 5226a94! I'm much more happy with the overall structure of the tutorials now. 

> I find the latter more explicit, and also if we insert tutorials in between it will force the iteration of all subsequent tutorials, which might create confusion if we refer to those in the forum (which we will)

I've found the [`:numbered:`](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html) option for `toctree::`, which will automatically number the sections. My hope is that this will be the best of both worlds (numbering, while still keeping explicit page names that won't have to be manually iterated when new tutorials are added).